tests/test_bytes.rs

```diff
@@ -1479,3 +1482,139 @@ fn split_to_empty_addr_mut() {
     let _ = &empty_start[..];
     let _ = &buf[..];
 }
+
+#[derive(Clone)]
+struct OwnedTester<const L: usize> {
+    buf: [u8; L],
+    drop_count: Arc<AtomicUsize>,
+    pub panic_as_ref: bool,
+}
+
+impl<const L: usize> OwnedTester<L> {
+    fn new(buf: [u8; L], drop_count: Arc<AtomicUsize>) -> Self {
+        Self {
+            buf,
+            drop_count,
+            panic_as_ref: false,
+        }
+    }
+}
+
+impl<const L: usize> AsRef<[u8]> for OwnedTester<L> {
+    fn as_ref(&self) -> &[u8] {
+        if self.panic_as_ref {
+            panic!("test-triggered panic in `AsRef<[u8]> for OwnedTester`");
+        }
+        self.buf.as_slice()
+    }
+}
+
+impl<const L: usize> Drop for OwnedTester<L> {
+    fn drop(&mut self) {
+        self.drop_count.fetch_add(1, Ordering::AcqRel);
```

Comment:
These tests are single-threaded, so the orderings don't matter. :woman_shrugging:

Reply:
True, but having these orderings littered everywhere isn't particularly readable, so I did some refactoring to tidy up into a `SharedAtomicCounter` struct.